common/src/service/HttpClientHelper.cs

```diff
@@ -307,12 +307,8 @@ private static async Task<T> ReadResponseMessageAsync<T>(HttpResponseMessage mes
                 return (T)(object)message;
             }
 
-#if NET451
-            T entity = await message.Content.ReadAsAsync<T>(token).ConfigureAwait(false);
-#else
             string str = await message.Content.ReadHttpContentAsStringAsync(token).ConfigureAwait(false);
-            T entity = JsonConvert.DeserializeObject<T>(str);
-#endif
+            T entity = JsonConvert.DeserializeObject<T>(str, new JsonSerializerSettings { DateParseHandling = DateParseHandling.None });
```

Comment:
By default, serializing/deserializing with Newtonsoft.Json will try to parse date-formatted strings to a date type, which drops trailing zeros in the microseconds date portion ("2023-01-31T10:37:08.45994**00**+01:00" -> "2023-01-31T10:37:08.45994+01:00"). By specifying `DateParseHandling` with `None`, the original string will be read as-is.

Reply:
Should we set the serializer settings at a library level so that all dates are handled as expected? (similar to how we set the max json depth value).